docs/reference-guides/core-blocks.md

```diff
@@ -276,7 +276,7 @@ Gather blocks in a layout container. ([Source](https://github.com/WordPress/gute
 -	**Name:** core/group
 -	**Category:** design
 -	**Supports:** align (full, wide), anchor, ariaLabel, color (background, gradients, link, text), spacing (blockGap, margin, padding), typography (fontSize, lineHeight), ~~html~~
--	**Attributes:** tagName, templateLock
+-	**Attributes:** alias, tagName, templateLock
```

Comment:
This is temporary. In the longer term we would add via Block Supports.

Reply:
Hi @andrewserong. Thanks for your input here - very much appreciated.

I've had a go at implementing Block Supports for `meta`. What I'm not clear on is should I also be registering block supports for `alias` or can meta now encompass anything?

Also I'm not clear on how we can automated the following code which is hard coded into the `core/group` block?

```js
__experimentalLabel( attributes, { context } ) {
		const { meta } = attributes;

		// In the list view, use the block's aliasattribute as the label.
		if ( context === 'list-view' && meta?.alias ) {
			return meta.alias;
		}
	},
```

I was thinking we'd register this label based on support for `alias`.

Reply:
Thanks for exploring this @getdave! I think I have an idea for how it could work. But, please feel free to back out of any of this if you think it's expanding the scope of this PR too much for the time being! I'm mostly excited to see how we can have this working across additional container blocks like Columns, but as a first step just getting it working for Group will have a good impact, I'm sure 🙂

The way the `addAttribute` callback works is that it adds an additional attribute if one doesn't exist when the block type is registered. I think we might be able to do much the same with generating a `__experimentalLabel` callback if one doesn't already exist. So something like the following pseudo-code (I'm typing out the idea but haven't tried it myself, apologies if it doesn't wind up working 😀):

```

/**
 * Filters registered block settings, adding an `__experimentalLabel` callback if one does not already exist.
 *
 * @param {Object} settings Original block settings.
 *
 * @return {Object} Filtered block settings.
 */
export function addLabelCallback( settings ) {
	// If labels provide their own label callback, do not override it.
	if ( settings.__experimentalLabel ) {
		return settings;
	}
	if ( hasBlockSupport( settings, 'meta' ) ) {
		settings.__experimentalLabel = ( attributes, { context } ) => {
			const { meta } = attributes;

			// In the list view, use the block's aliasattribute as the label.
			if ( context === 'list-view' && meta?.alias ) {
				return meta.alias;
			}
		};
	}

	return settings;
}


addFilter(
	'blocks.registerBlockType',
	'core/meta/addLabelCallback',
	addLabelCallback
);
```

> What I'm not clear on is should I also be registering block supports for alias or can meta now encompass anything?

What if it works like some other block supports like Layout or Border where if you set the root `meta` to `true` then all meta features are enabled, but you can also set `meta` to an object with `alias` set to `true` to just opt-in to that one feature? Here's where `hasBorderSupport` is defined which has support for a root boolean value or an object: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/border.js#L314

---

One other thought, if we move to storing the value under the `meta` object, would it be worth renaming `alias` to a name that's more specific to its usage, something like `customBlockLabel`, `blockNameAlias` or `blockAlias`?

Reply:
> I added a separate comment, but I was wondering if it's worth having an attribute that is a container for any kind of in-editor metadata, rather than a dedicated alias attribute?

See https://github.com/WordPress/gutenberg/pull/40393 from @jorgefilipecosta where there is a similar discussion about using `metadata` in attributes to store details specific only to the experience in the editor.

There are other existing attributes falling under the same category. They are used as top-level attributes and don't impact the front end at all:
- `lock` available for all blocks ([always added with PHP](https://github.com/WordPress/wordpress-develop/blob/c1cf07e4e30cf0a2fde3f2be0a6983025b3d2062/src/wp-includes/class-wp-block-type.php#L213-L215) and also fully controlled through a JS hook)
- `templateLock` used with Column, Cover, and Group blocks
- `allowedBlocks` used with Column and Group blocks

By the way, there is also the `settings` attribute used for stuff related to `theme.json` implemented in https://github.com/WordPress/gutenberg/pull/40547.

It would be great to come up with a general solution that works for this type of control values that are completely useless when rendering the output of the block. It definitely is tricky to continue adding them as top-level attributes because you never know whether custom blocks or plugins extending blocks didn't use the same names for a different purpose. We encountered similar challenges with core blocks where some blocks used `align` for block alignment but others for text alignment.

Reply:
> See https://github.com/WordPress/gutenberg/pull/40393 from @jorgefilipecosta where there is a similar discussion about using metadata in attributes to store details specific only to the experience in the editor.

Excellent, thanks @gziolo, great to see this has already been explored, looks like there's a lot of overlap here 🙂

Reply:
By the way, the downside of using an object for an attribute like `meta` is that maintaining its value is no longer simple:
- when updating a field in the `meta` attribute, you always have to ensure other metadata is correctly stored
- when using block variations, if you want to override one of the `meta` fields then you need to remember to copy all other settings
- similar concern applies to block patterns in the case when the default values for fields in `meta` change, if the pattern changes any value, then it also seals other default values

Well, you still could have some strategies for mitigating that, but it's more difficult to reason where every attribute is handled individually, and you have control over every default value and attribute type.

Reply:
Let's definitely reconcile with #40393 as the semantic surface needs to be very clear, otherwise things will get confusing for people and for maintenance.

Reply:
Great info here thanks all. I'll drop in on https://github.com/WordPress/gutenberg/pull/40393 to seek to align our implementations.